Modules/Tracker/QuestieTrackerPrivates.lua

```diff
@@ -8,56 +8,335 @@ local QuestieCombatQueue = QuestieLoader:ImportModule("QuestieCombatQueue")
+            -- your player turn :(
+            if not IsMouselooking() then
+                MouselookStart()
+                    mouselookTicker = C_Timer.NewTicker(0.1, function()
+                    if not IsMouseButtonDown(button) then
+                        MouselookStop()
+                        mouselookTicker:Cancel()
+                    end
+                end)
+            end
         end
     end
 end
 
-function _QuestieTracker:OnDragStop()
-    Questie:Debug(DEBUG_DEVELOP, "[_QuestieTracker:OnDragStop]")
-    if not startDragPos or not startDragPos[4] or not startDragPos[5] or not endDragPos or not startDragAnchor then
+function _QuestieTracker:OnDragStop(button)
+    Questie:Debug(DEBUG_DEVELOP, "[_QuestieTracker:OnDragStop]", button)
+
+    if IsMouseButtonDown(button) or not startDragPos or not startDragPos[4] or not startDragPos[5] or not endDragPos or not startDragAnchor then
         return
     end
+
     local baseFrame = QuestieTracker:GetBaseFrame()
     _QuestieTracker.isMoving = false
-
     endDragPos = {baseFrame:GetPoint()}
     baseFrame:StopMovingOrSizing()
 
+    -----------------------------------------------------------------------------------------------
+    -- Since frame:SetClampedToScreen() only applies to frames, the end X,Y values can potentially
+    -- traverse off the players screen and cause weird anomalies such as frame "snapping" where
+    -- values from screen edge and (mouse pointer/frame set point) end locations can be located
+    -- beyond the players screen values. The following checks simply subtracts those values keeping
+    -- the end point setting inside the players screen boundaries, thus preventing frame "snapping"
+    -- from ever occurring. Screen coords are calculated based on the relativePoint of the frame.
+    --
+    -- Before frame:StartMoving() is called the relativePoint of frame:GetPoint() is, in Questie's
+    -- case, "CENTER", so (-x, -y) values would be in the "BOTTOMLEFT" area of the players screen.
+    --
+    -- After frame:StartMoving() or frame:StopMovingOrSizing() is called the relativePoint of
+    -- frame:GetPoint() is, in Questie's case, "TOPLEFT", so (0, 0) values would set the frame in
+    -- the "TOPLEFT" most corner of the players screen.
+    -----------------------------------------------------------------------------------------------
+
+    -- Trim X end cords
+    local maxLeft = 0
+    if endDragPos[4] < maxLeft then
+       endDragPos[4] = maxLeft
+    end
+
+    local maxRight = GetScreenWidth() - baseFrame:GetWidth()
+    if endDragPos[4] > maxRight then
+        endDragPos[4] = maxRight
+    end
+
+    -- Trim Y end cords
+    local maxBottom = -GetScreenHeight() - -baseFrame:GetHeight()
+    if endDragPos[5] < maxBottom then
+        endDragPos[5] = maxBottom
+    end
+
+    local maxTop = 0
+    if endDragPos[5] > maxTop then
+        endDragPos[5] = maxTop
+    end
```

Comment:
You added a comment perfect for a function name. Maybe move it into one?

Reply:
Fixed in latest commit